catapi/views/cat_view.py

```diff
@@ -14,14 +16,18 @@
     response_model=dto.Cat,
     status_code=status.HTTP_201_CREATED,
 )
-async def create_cat(unsaved_cat: dto.UnsavedCat) -> dto.JSON:
+async def create_cat(
+    unsaved_cat: dto.UnsavedCat,
+    background_tasks: BackgroundTasks,
+) -> dto.JSON:
     """
     Create view for creating a new Cat given an UnsavedCat payload.
 
     \f
     :return:
     """
     cat = await cat_domain.create_cat(unsaved_cat)
+    background_tasks.add_task(cat_events.fire_cat_created, cat_id=cat.dict().get("id"))
```

Comment:
[question] What will this do and how will this result in a message "cat.created" being put in the RabbitMQ queue?

Reply:
@sanjeevpc1984 The intention is to use rabbitMQ to have consumer invoke this method instead of invoking this directly from here.